### ISSUE TITLE
Add method to assert that that a message is present in p:messages

### ISSUE THREAD

**Issue by krichter722** [omnifaces/arquillian-primefaces#7]
It'd be nice to have a method which checks that a messages added with `FacesContext.addMessage` is present in a `p:messages`.

**Comment by BalusC** [comment]
```
@FindBy(id="messages")
private WebElement messages;

@Test
public void test() {
    Assert.assertTrue(!messages.getText().isEmpty());
}
```

Could it be simpler? Maybe an `assertHasText(WebElement)` might be useful, but that won't reduce much boilerplate code.

**Comment by krichter722** [comment]
That randomly fails because `messages.getText` sometimes only contains the summary whereas a screenshot confirms that the text content should be present (with `showDetails= "true"` of course). I figured that there might be something to pay attention to which I don't know about, but maybe it's just a bug (my attempts to create a SSCCE were not successful so far).

I guess if you say that `messages.getText()` should work based on your insight on PrimeFaces internals, then this request is invalid and you can close it (I'll have to extract a SSCCE and file a bug, then).

You could still add a method doing the above in order to make clear that that's how `p:messages` ought to be evaluated.

It'd be nice to have a possibility to check the presence of multiple messages without the need to account for the order, though. This requires inspection of the `li` elements and I guess it's worth to have a method for it.

**Comment by BalusC** [comment]
If the message is updated by ajax, then there may indeed be a timing error. You'd best wait for the element to have text, if a message is expected.

Something like this:

```
Graphene.waitGui().withTimeout(5, TimeUnit.SECONDS).until(browser -> !messages.getText().trim().isEmpty());
```

Let me know if that solves your intermittent problem.

**Comment by krichter722** [comment]
I tried waiting for non-empty text as well as expected text. After switching from Selenium (Graphene?) driver `phantomjs` to `chromeheadless` I passed more than 10 error-free runs which makes me believe that this was an issue with `phantomjs` (trace logging innerHTML of the `WebElement` corresponding to `p:messages` reveals that the text is indeed the summary only (randomly)). This means that this issue is invalid for me.

So far, I've been working with https://gitlab.com/krichter/selenium-tools/blob/master/src/main/java/richtercloud/selenium/tools/SeleniumHelper.java#L363 and https://gitlab.com/krichter/selenium-tools/blob/master/src/main/java/richtercloud/selenium/tools/SeleniumHelper.java#L402 which allow to wait for a set of expected texts which helps to account for different order of messages inside `p:messages`. Maybe you want to take a look and incorporate this (partly) here.
